src/transformers/hf_argparser.py

```diff
@@ -124,6 +126,8 @@ def _add_dataclass_arguments(self, dtype: DataClassType):
                 ), "{} cannot be a List of mixed types".format(field.name)
                 if field.default_factory is not dataclasses.MISSING:
                     kwargs["default"] = field.default_factory()
+                else:
+                    kwargs["required"] = True
```

Comment:
Actually this looks like it ignores the default if present, so it should be
```suggestion
                elif field.default is dataclasses.MISSING:
                    kwargs["required"] = True
```

Reply:
Oh yeah, sure.